src/main/java/duke/io/Parser.java

```diff
@@ -0,0 +1,198 @@
+package duke.io;
+
+import duke.task.DukeException;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public class Parser {
+    private final Layout layout;
+    
+    public Parser() {
+        this.layout = new Layout();
+    }
+    
+
+    /**
+     * Return date and description of task from user input.
+     * 
+     * @param arr Array of words in user input.
+     * @return Array of date and description.
+     */
+    public String [] getDateAndDescription(String [] arr) {
+        boolean reached = false;
```

Comment:
Maybe rename reached to reached[Something] so that it will be easier to understand?